### PR TITLE
Add Mox.get_executed_calls/1

### PR DESCRIPTION
**Pull request by whatyouhide**
Kicking off some code for #133.

I have no clue what I'm doing, I'm not familiar with Mox's codebase so I'm just sort of winging it.

### CODE REVIEW

File: lib/mox.ex [hunk 1]
```diff
@@ -812,7 +817,7 @@ defmodule Mox do
   def __dispatch__(mock, name, arity, args) do
     all_callers = [self() | caller_pids()]
 
-    case Mox.Server.fetch_fun_to_dispatch(all_callers, {mock, name, arity}) do
+    case Mox.Server.fetch_fun_to_dispatch(all_callers, {mock, name, arity}, args) do
```
Comment (josevalim) [inline lib/mox.ex:820]: This means we are copying all argument data to a separate process. :(

File: lib/mox.ex [hunk 2]
```diff
@@ -763,6 +763,11 @@ defmodule Mox do
     verify_mock_or_all!(self(), mock, :test)
   end
 
+  def get_executed_calls(mock) do
+    validate_mock!(mock)
+    Mox.Server.get_executed_calls(mock)
```
Comment (josevalim) [inline lib/mox.ex:768]: Also, is this safe to execute concurrently? It doesn't look like it is the case. :(

File: lib/mox.ex (hunk 1 above)
Comment (whatyouhide) [reply lib/mox.ex:820]: Right, is that an issue in tests? We would do the same when using the `send/2` trick to send calls to the parent process.

Any ideas on how to do this without copying the args?

File: lib/mox.ex (hunk 2 above)
Comment (whatyouhide) [reply lib/mox.ex:768]: It definitely isn't 😉 Will have to figure it out with the allowance.

**Comment by whatyouhide** [comment]
@josevalim this is just to get the ball rolling. Are we interested in this feature? If so, we can try to hash out the things that are borked 🥦

File: lib/mox.ex (hunk 1 above)
Comment (josevalim) [reply lib/mox.ex:820]: the problem is that this enables this for everything by default. We would need to store it somewhere if we want to collect or not.

**Pull request closed without merging**